skbio/stats/composition.py

```diff
@@ -608,10 +610,271 @@ def centralize(mat):
+    prop_cut = np.zeros_like(cutoff, dtype=np.float32)
+    for cut in range(len(cutoff)):
+        prop_cut[cut] = sum(W > n_feat*cutoff[cut]) / len(W)
+    for i in range(len(cutoff)-1):
+        dels[i] = abs(prop_cut[i] - prop_cut[i+1])
+
+    if (dels[1] < tau) and (dels[2] < tau) and (dels[3] < tau):
+        nu = cutoff[1]
+    elif (dels[1] >= tau) and (dels[2] < tau) and (dels[3] < tau):
+        nu = cutoff[2]
+    elif (dels[2] >= tau) and (dels[3] < tau) and (dels[4] < tau):
+        nu = cutoff[3]
+    else:
+        nu = cutoff[4]
+    reject = (W >= nu*n_feat)
+    return pd.Series(W, index=labs), pd.Series(reject, index=labs)
+
+
+def _check_composition(x, ignore_zeros=True):
+    """
+    Checks to make sure that composition meets the mininum criteria
+    Also casts composition into a pandas dataframe
+
+    Parameters
+    ----------
+    x : array_like or pd.DataFrame
+       Input composition matrix
+       where rows=samples and columns=features
+    ignore_zeros : bool
+       Specifies if compositions are allowed to have zeros
+
+    Returns
+    -------
+    pd.DataFrame
+       Validated composition matrix
+    """
+    if x.ndim > 2:
+        raise ValueError("Input matrix can only have two dimensions or less")
+    if isinstance(x, pd.DataFrame):
+        mat = closure(x)
+        samp_labs = x.index
+        feat_labs = x.columns
+    else:
+        mat = np.atleast_2d(x)
+        r, c = mat.shape
+        samp_labs = range(r)
+        feat_labs = range(c)
+    if np.any(mat == 0) and not ignore_zeros:
+        raise ValueError('Cannot handle zeros in compositions. '
+                         'Make sure to run a zero replacement method')
+    return pd.DataFrame(mat, index=samp_labs, columns=feat_labs)
+
+
+def _holm(p):
+    """
+    Performs Holm-Boniferroni correction for pvalues
+    to account for multiple comparisons
+
+    Parameters
+    ---------
+    p: numpy.array
+        array of pvalues
+
+    Returns
+    -------
+    numpy.array
+        corrected pvalues
+    """
+    K = len(p)
+    sort_index = -np.ones(K, dtype=np.int64)
+    sorted_p = np.sort(p)
+    sorted_p_adj = sorted_p*(K-np.arange(K))
+    for j in range(K):
+        idx = (p == sorted_p[j]) & (sort_index < 0)
+        num_ties = len(sort_index[idx])
+        sort_index[idx] = np.arange(j, (j+num_ties), dtype=np.int64)
+
+    sorted_holm_p = [min([max(sorted_p_adj[:k]), 1])
+                     for k in range(1, K+1)]
+    holm_p = [sorted_holm_p[sort_index[k]] for k in range(K)]
+    return holm_p
+
+
+def _log_compare(mat, cats,
+                 stat_func=scipy.stats.ttest_ind):
+    """
+    Calculates pairwise log ratios between all otus
+    and performs a permutation tests to determine if there is a
+    significant difference in OTU ratios with respect to the
+    variable of interest
+
+    Parameters
+    ----------
+    mat: np.array
+       rows = samples
+       columns = features (i.e. OTUs)
+    cat: np.array, float
```

Comment:
I'm fine with this and `mat` being abbreviated as it is private, but `cat` should be `cats` to match the function sig.